test/e2e_node/resource_metrics_test.go

```diff
@@ -89,15 +89,19 @@ var _ = framework.KubeDescribe("ResourceMetricsAPI", func() {
 					fmt.Sprintf("%s::%s::%s", f.Namespace.Name, pod1, "busybox-container"): boundedSample(10*e2evolume.Kb, 80*e2evolume.Mb),
 				}),
 			})
+			ginkgo.By("ensuring both resource metrics APIs return the same values")
 			ginkgo.By("Giving pods a minute to start up and produce metrics")
-			gomega.Eventually(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchV1alpha1Expectations)
+			gomega.Eventually(getResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
+			gomega.Eventually(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
 			ginkgo.By("Ensuring the metrics match the expectations a few more times")
-			gomega.Consistently(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchV1alpha1Expectations)
+			gomega.Consistently(getResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
+			gomega.Consistently(getV1alpha1ResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
```

Comment:
I don't know ginkgo well enough to know if we can do two comparisons in one call here. 

This serializes everything, so it takes 2x as long since we wait a minute for each.

Reply:
I'm fine with this, or i'm also fine having separate `If(` blocks, and making the entire test a function that takes the getResourceMetrics() function.  The advantage of the latter is that it is easy to tell which endpoint isn't working.

Reply:
I'll mess around with it some more and see if I can split it up without introducing a bunch of duplication.

Reply:
I need to modify the matcher to for each case to ignore the other's pods. Not sure if there is a namespaced call to resource API that this should use instead.